LifeCycleAssessment_Engine/Query/GetElementScope.cs

```diff
@@ -59,13 +59,19 @@ public static ScopeType GetElementScope(this IElementM elementM)
             // get scope fragment
             Scope scope = fragments.Select(x => x).Where(y => y is Scope).FirstOrDefault() as Scope;
 
-            ScopeType type = scope.ScopeType;
-
             if (scope == null)
             {
                 return ScopeType.Undefined;
             }
 
+            // For cases where objects have multiple fragments, but don't have a scope fragment
+            if (fragments.Count >= 1 && scope == null)
```

Comment:
If the `scope` is null, then it will already have been caught by the previous `if` statement, so the check here with the `&& scope == null` is redundant. I'd advise either removing the first `null` check, or moving the `fragments.Count >= 1` part of the check into the first check and removing this new one.

Reply:
Yeah got a little trigger happy to make it into 5.1 with the fix (which does work) but gave too little thought on code redundancy. Thanks for the review! Will pick this on up in the next milestone.

Reply:
I've tried to resolve the issue you mentioned in the code, as well as elaborated on the original issue. Please let me know if there are any additional questions, or if you'd like help testing this specific PR.